src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -459,6 +460,7 @@ def preprocess(self, inputs, chunk_length_s=0, stride_length_s=None):
     def _forward(self, model_inputs, return_timestamps=False, **generate_kwargs):
         attention_mask = model_inputs.pop("attention_mask", None)
         stride = model_inputs.pop("stride", None)
+        segment_size = model_inputs.pop("segment_size", None)
```

Comment:
Are there any cases when stride is None and we want to use `segment_size`? 

Because we're popping both from `model_inputs` here, technically, the user can pass this in directly as an argument when calling `pipeline.forward`. In which case, it would be good to have input validation on these argument

Reply:
We will get stride is None when using this pipeline: 

```
pipe = pipeline(
            task="automatic-speech-recognition",
            model="openai/whisper-large-v3",
            return_timestamps="word",
        )
```

--> `segment_size` is the alternative to compute and add `num_frames` to the generate_kwargs to be passed to the generate method [here](https://github.com/huggingface/transformers/blob/15395df9d11d172197a69233418f0c49d7f00980/src/transformers/pipelines/automatic_speech_recognition.py#L495).

Reply:
Sorry, I mistyped. I meant when stride is not None

Reply:
We want to use `segment_size` only when stride is None.

Reply:
I've added the following lines: 

```
if stride is not None and segment_size is not None: 
            raise ValueError("segment_size must be used only when stride is None")
```

Is it ok ?